flake.nix

```diff
@@ -279,9 +280,13 @@
         enableParallelBuilding = true;
 
         configureFlags = testConfigureFlags; # otherwise configure fails
-        dontBuild = true;
         doInstallCheck = true;
 
+        buildPhase = ''
+          # Remove the source files to make sure that we're not accidentally rebuilding Nix
+          rm src/**/*.cc
+        '';
+
```

Comment:
Can we instead use a filtered source like @roberth did? In other words whitelisting not blacklisting, and getting some free rebuild-avoidance.

(The latter is useful for the thing that @benradf and I did were we wanted to change the test suite without rebuilding Nix and run it in derivation.)

Reply:
> Can we instead use a filtered source like @roberth did? In other words whitelisting not blacklisting, and getting some free rebuild-avoidance.

I thought about that, but since these derivations depend on a local Nix any way, we're bound to rebuild it if we make any change that requires rebuilding Nix, so it didn't feel worth the hassle. Also, the build system makes it a bit painful to really know what is required, so I'd rather not go down that way until we change it ;)

Reply:
That is why I included the parenthetical: I believe the sort of thing @benradf  did where we change the tests but not Nix itself, and a clean rebuild, is bound to be needed again, and should be easier. If we can filter the sources out of this, and the tests out of building Nix, then we have that easy.

> Also, the build system makes it a bit painful to really know what is required, so I'd rather not go down that way until we change it ;)

Yes, but it is a one-time cost. @roberth had some good stuff written up how blacklists just silent regress in that as more things creep in, the rebuilding benefits get less and less, but very slowly in a hard-to-notice way. Whitelists are hard to do the first time, but then simple to maintain because there is no silent regressions, and explicit breakage is usually easy to fix the moment it occurs.

Reply:
> That is why I included the parenthetical: I believe the sort of thing @benradf did where we change the tests but not Nix itself, and a clean rebuild, is bound to be needed again, and should be easier. If we can filter the sources out of this, and the tests out of building Nix, then we have that easy.

I'm not sure I follow: this depends on the `nix` derivation (which depends on the source tree obviously) and the test folder, so if you change any of these, you _will_ have to rebuild the tests, right? The interesting thing would be to filter out the `test` folder when building Nix, but that's a different thing

> blacklists just silent regress in that as more things creep in, the rebuilding benefits get less and less, but very slowly in a hard-to-notice way

They are indeed, but I don't really care here :) All I want is to make sure that we're not rebuilding Nix. Whether we access anything we shouldn't is something that would be great to have, but not the concern of this PR (and not a priority imho, there are far more useful things to work on).

Reply:
I'll gladly let you work on this as a follow-up then :stuck_out_tongue:

Reply:
I'll do it if the next time you do the whitelist from beginning, deal? :)